RetailCoder.VBE/Inspections/QuickFixes/AssignedByValParameterMakeLocalCopyQuickFix.cs

```diff
@@ -99,11 +102,23 @@ private void ReplaceAssignedByValParameterReferences(IModuleRewriter rewriter, s
 
         private void InsertLocalVariableDeclarationAndAssignment(IModuleRewriter rewriter, string localIdentifier)
         {
-            var content = Tokens.Dim + " " + localIdentifier + " " + Tokens.As + " " + _target.AsTypeName + Environment.NewLine
-                + (_target.AsTypeDeclaration is ClassModuleDeclaration ? Tokens.Set + " " : string.Empty)
-                + localIdentifier + " = " + _target.IdentifierName;
+            var content = Tokens.Dim + " " + localIdentifier + " " + Tokens.As + " " + _target.AsTypeName + Environment.NewLine;
+            string assignmentFormat = "{0} = {1}";
+            if (RequiresSetAssignment(_target))
+            {
+                assignmentFormat = "Set {0} = {1}";
+            }
 
+            content = content + string.Format(assignmentFormat, localIdentifier, _target.IdentifierName);
             rewriter.InsertBefore(((ParserRuleContext)_target.Context.Parent).Stop.TokenIndex + 1, "\r\n" + content);
```

Comment:
Actually, it is probably being compiled to use a `StringBuilder` behind the scenes.